src/main/java/org/folio/inventory/support/KafkaConsumerVerticle.java

```diff
@@ -0,0 +1,165 @@
+  private HttpClient httpClient;
+  private Storage storage;
+
+  @Override
+  public void stop(Promise<Void> stopPromise) {
+    var stopFutures = consumerWrappers.stream()
+      .map(KafkaConsumerWrapper::stop)
+      .toList();
+
+    GenericCompositeFuture.join(stopFutures)
+      .onComplete(ar -> stopPromise.complete());
+  }
+
+  protected abstract Logger getLogger();
+
+  protected KafkaConsumerWrapper<String, String> createConsumer(String eventType) {
+    return createConsumer(eventType, true);
+  }
+
+  protected KafkaConsumerWrapper<String, String> createConsumer(String eventType, boolean namespacedTopic) {
+    var kafkaConsumerWrapper = KafkaConsumerWrapper.<String, String>builder()
+      .context(context)
+      .vertx(vertx)
+      .kafkaConfig(getKafkaConfig())
+      .loadLimit(getLoadLimit())
+      .globalLoadSensor(new GlobalLoadSensor())
+      .subscriptionDefinition(getSubscriptionDefinition(getKafkaConfig().getEnvId(), eventType, namespacedTopic))
+      .build();
+    consumerWrappers.add(kafkaConsumerWrapper);
+    return kafkaConsumerWrapper;
+  }
+
+  protected KafkaConfig getKafkaConfig() {
+    if (isNull(kafkaConfig)) {
+      kafkaConfig = KafkaConfig.builder()
+        .envId(getConfig().getString(KAFKA_ENV))
+        .kafkaHost(getConfig().getString(KAFKA_HOST))
+        .kafkaPort(getConfig().getString(KAFKA_PORT))
+        .okapiUrl(getConfig().getString(OKAPI_URL))
+        .replicationFactor(parseInt(getConfig().getString(KAFKA_REPLICATION_FACTOR)))
+        .maxRequestSize(parseInt(getConfig().getString(KAFKA_MAX_REQUEST_SIZE)))
+        .build();
+      getLogger().info("kafkaConfig: {}", kafkaConfig);
+    }
+    return kafkaConfig;
+  }
+
+  protected HttpClient getHttpClient() {
+    if (isNull(httpClient)) {
+      httpClient = vertx.createHttpClient();
+    }
+    return httpClient;
+  }
+
+  protected Storage getStorage() {
+    if (isNull(storage)) {
+      storage = Storage.basedUpon(getConfig(), vertx.createHttpClient());
+    }
+    return storage;
+  }
+
+  protected MappingMetadataCache getMappingMetadataCache() {
+    return MappingMetadataCache.getInstance(vertx, getHttpClient());
+  }
+
+  protected ProfileSnapshotCache getProfileSnapshotCache() {
+    if (isNull(profileSnapshotCache)) {
+      var profileSnapshotExpirationTime = getCacheEnvVariable(PROFILE_SNAPSHOT_CACHE_EXPIRATION_TIME);
+      profileSnapshotCache = new ProfileSnapshotCache(vertx, getHttpClient(), Long.parseLong(profileSnapshotExpirationTime));
+    }
+    return profileSnapshotCache;
+  }
+
+  protected String getCacheEnvVariable(String variableName) {
+    var cacheExpirationTime = getConfig().getString(variableName);
+    if (isBlank(cacheExpirationTime)) {
+      cacheExpirationTime = CACHE_EXPIRATION_DEFAULT;
+    }
+    return cacheExpirationTime;
+  }
+
+  protected int getMaxDistributionNumber() {
+    return getConsumerProperty(MAX_DISTRIBUTION_NUMBER_TEMPLATE, MAX_DISTRIBUTION_NUMBER_DEFAULT);
```

Comment:
I think calling getConsumerProperty  will cause prolmens :( 
In `getConsumerProperty`, we remove the word "Verticle" from the class name. However, if you look at the (now removed) getMaxDistributionNumber  methods in `DataImportConsumerVerticle` you can see that the system property used is `inventory.kafka.DataImportConsumer**Verticle**.maxDistributionNumber`

Reply:
it's true... to make it safer, we'd better pass a specific value each time. Changed my approach. Thank you!